antarest/study/storage/utils.py

```diff
@@ -383,3 +394,83 @@ def ignore_outputs(directory: str, _: t.Sequence[str]) -> t.Sequence[str]:
         study.tree.denormalize()
         duration = "{:.3f}".format(time.time() - stop_time)
         logger.info(f"Study '{study_dir}' denormalized in {duration}s")
+
+
+def is_folder_safe(workspace: WorkspaceConfig, folder: str) -> bool:
+    """
+    Check if the provided folder path is safe to prevent path traversal attack.
+
+    Args:
+        workspace: The workspace name.
+        folder: The folder path.
+
+    Returns:
+        `True` if the folder path is safe, `False` otherwise.
+    """
+    requested_path = workspace.path / folder
+    safe_dir = os.path.realpath(workspace.path)
+    real_requested_path = os.path.realpath(requested_path)  # resolve symbolic links like ~ , .. and .
+
+    common_prefix = os.path.commonprefix(
+        (os.path.realpath(real_requested_path), safe_dir)
+    )  # get the common prefix of the two paths
+    return common_prefix == safe_dir  # if requested path isn't in workspace, then common prefix is different
+
+
+def is_study_folder(path: Path) -> bool:
+    path.relative_to
+    return path.is_dir() and (path / "study.antares").exists()
+
+
+def is_aw_no_scan(path: Path) -> bool:
+    if (path / "AW_NO_SCAN").exists():
+        logger.info(f"No scan directive file found. Will skip further scan of folder {path}")
```

Comment:
Remove this log and do it inside one line as you're already logging this after calling this method